Program.cs

```diff
@@ -1,4 +1,128 @@
-﻿using System;
-Console.WriteLine ("Hello, World!");
+﻿// --------------------------------------------------------------------------------------------
+// Training ~ A training program for interns at Metamation, Batch - July 2023
+// Copyright (c) Metamation India.                                              
+// ------------------------------------------------------------------------
+// Program.cs
+// MYLIST<T> 
+// Implement a custom MyList<T> class using arrays as the underlying data structure.
+// --------------------------------------------------------------------------------------------
+using System;
+// Test cases
+MyList<int> list = new ();
+list.Add (1);
+list.Add (2);
+list.Add (3);
+list.Add (4);
+list.Add (5);
+list.Add (6);
+list.Add (7);
+list.Display ();
+list.Remove (7);
+list.Clear ();
+list.Display ();
+list.Insert (0, 1);
+list.Insert (1, 2);
+list.Insert (2, 3);
+list.Insert (3, 4);
+list.Insert (4, 5);
+list.Insert (5, 6);
+list.Display ();
+list.RemoveAt (0);
+list.RemoveAt (1);
+list.Display ();
 
-   
\ No newline at end of file
+/// <summary>Create a class</summary>
+/// <typeparam name="T">Datatype of the list</typeparam>
+public class MyList<T> {
+   private T[] mArray;
+   private int mCount;
+   /// <summary>Creating a Constructor for initializing the values</summary>
+
+   public MyList () {
+      // Initialize the array capacity as 4. 
+      mArray = new T[4];
+      mCount = 0;
+   }
+   /// <summary>Counting the elememt in the list</summary>
+
+   public int Count => mCount;
+   /// <summary>Calculating capacity of the array</summary>
+
+   public int Capacity => mArray.Length;
+   /// <summary>Set and return the value based on the index</summary>
+   /// <param name="index">Index to be change</param>
+   /// <returns>Changed value of the index</returns>
+   /// <exception cref="IndexOutOfRangeException">When the index is out of range</exception>
+
+   public T this[int index] {
+      get {
+         if (index < 0 || index >= mCount) throw new IndexOutOfRangeException ("Index is out of the valid range.");

```

Comment:
This line of code to throw `IndexOutOfRangeException` is used more than once. Move it to a separate function.